src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -106,3 +139,27 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
 
     return nodejsLaunchConfig
 }
+
+function dirTypeScriptFiles(base:string, files?:string[], result?:string[]): string[] {
+    files = files || readdirSync(base) 
+    result = result || [] 
+
+    files.forEach( 
+        function (file) {
+            const newbase = path.join(base, file)
+            if ( statSync(newbase).isDirectory() )
+            {
+                result = dirTypeScriptFiles(newbase, readdirSync(newbase), result)
+            }
+            else
+            {
+                if ( file.substr(-3) === '.ts')
+                {
+                    result!.push(newbase)
```

Comment:
`result!` is needed i guess because it was being overwritten above. after fixing that, don't need `!` here.